bugbot/crash/analyzer.py

```diff
@@ -477,6 +477,41 @@ def is_near_allocator_related_crash(self) -> bool:
         """
         return self.is_near_allocator_crash or self.is_potential_near_allocator_crash
 
+    @property
+    def is_content_crash(self) -> bool:
+        """Whether the crash is related to content process."""
+        for row in self.signature["facets"]["process_type"]:
+            if row["term"].lower() == "content":
+                return row["count"] > 0
+        return False
+
+    @property
+    def is_gpu_crash(self) -> bool:
+        """Whether the crash is related to GPU."""
+        for row in self.signature["facets"]["process_type"]:
+            if row["term"].lower() == "gpu":
+                return row["count"] > 0
+        return False
+
+    @property
+    def is_parent_crash(self) -> bool:
+        """Whether the crash is related to the parent process."""
+        for row in self.signature["facets"]["process_type"]:
+            if row["term"].lower() == "parent":
+                return row["count"] > 0
+        return False
+
+    @property
+    def crash_kind(self) -> str:
+        """The kind of the crash based on the crashing process type."""
+        if self.is_plugin_crash:
```

Comment:
Why is this structured like this rather than iterating over the process types once and then having a case statement or something in the body of the loop?

Reply:
I formed it like this to give priority based on the types, instead of the frequency. So the kind will be "Plugin" even if there is one plugin crash and 100 GPU crashes. I'm not sure this is the desirable output. The motivation for this was to be consistent with Soccor, similar to the case when it decided between "Browser Crash" and "Plugin Crash"

Reply:
Okay, that's interesting. Part of my concern here was about quadratic behavior but maybe that's not particularly important. As for a priority, my preference would be to follow whatever logic Socorro uses to determine the value for the "Process Type" field in an individual crash report. I don't know what exactly it does, but I think it has stood the test of time for producing something useful. It would also be confusing for people if the output here ended up being different from what the crash report says. So consistency has a value, too.